lib/linguist/heuristics.yml

```diff
@@ -453,16 +453,15 @@ disambiguations:
     pattern: '(?i:^\\i\b|AS \$\$|LANGUAGE ''?plpgsql''?|BEGIN( WORK )?;)'
   # IBM db2
   - language: SQLPL
-    pattern: "(?i:alter module|MODE DB2SQL)"
+    pattern: '(?i:alter module|MODE DB2SQL|\bSYSCAT\.\b|\bSYSPROC\.\b|ASSOCIATE RESULT SET|\bEND!\s*$)'
```

Comment:
1. Since the pattern's case-insensitive, we should uppercase `alter module` so it's consistent with the rest of the pattern. To a casual reader, the discrepancy might be mistaken as deliberate.

2. `\.\b|` is a little vague about its intent, because `\.` is already a word-`\b`oundary, and nothing follows to indicate which sort of character is expected after it. For example `\.\b` would match `A.B`, but not `A. `. If in doubt, remove it entirely.

Suggested replacement:

~~~diff
-pattern: '(?i:alter module|MODE DB2SQL|\bSYSCAT\.\b|\bSYSPROC\.\b|ASSOCIATE RESULT SET|\bEND!\s*$)'
+pattern: '(?i:ALTER +MODULE|MODE +DB2SQL|\bSYS(CAT|PROC)\.|ASSOCIATE +RESULT +SET|\bEND!\s*$)'
~~~

Also, see my below remarks about ` +` being replaced by `\s+` instead.

Reply:
Added  your suggestions (with `\s+`).